### PR TITLE
Implement RpcHandler::GetWriter and add unittest

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer_grpc/framework/server_test.cc [hunk 1]
```diff
@@ -173,6 +192,31 @@ TEST_F(ServerTest, ProcessBidiStreamingRpcTest) {
...
+  proto::GetEchoResponse result;
+  proto::GetEchoRequest request;
+  request.set_input(13);
+
+  Server* server = server_.get();
+  std::thread response_thread([server]() {
+    std::future<EchoCallback> writer_future =
+        server->GetContext<MathServerContext>()->echo_writer_.get_future();
+    writer_future.wait();
+    CHECK(writer_future.get()());
```
Comment (gaschler) [inline cartographer_grpc/framework/server_test.cc]: Assign the callback to a variable in a separate line so this becomes more readable.

File: cartographer_grpc/framework/server_test.cc [hunk 2]
```diff
@@ -28,9 +30,11 @@ namespace cartographer_grpc {
 namespace framework {
 namespace {
 
+using EchoCallback = std::function<bool()>;
```
Comment (gaschler) [inline cartographer_grpc/framework/server_test.cc]: naming: This is a writer action with a captured response, name it   WriteResponseFunction, WriteResponseCallback?

Comment (cschuet) [reply cartographer_grpc/framework/server_test.cc]: Renamed to `EchoResponder`.

File: cartographer_grpc/framework/server_test.cc (hunk 1 above)
Comment (cschuet) [reply cartographer_grpc/framework/server_test.cc]: Done.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 2.

**Pull request merged**